File: src/main/java/xyz/oribuin/lilori/commands/CmdPrefix.java
package xyz.oribuin.lilori.commands;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.Permission;
import xyz.oribuin.lilori.managers.command.Command;
import xyz.oribuin.lilori.managers.command.CommandEvent;

import java.awt.*;
import java.util.Collections;

public class CmdPrefix extends Command {

    public CmdPrefix() {
        this.name = "Prefix";
        this.description = "Change the bot permission";
        this.aliases = Collections.emptyList();
        this.arguments = Collections.singletonList("<prefix>");
        //this.userPermissions = new Permission[]{Permission.ADMINISTRATOR};
    }

    @Override
    public void executeCommand(CommandEvent event) {
        String[] args = event.getMessage().getContentRaw().split(" ");

        if (args.length == 1) {
            event.reply(event.getAuthor().getAsMention() + ", Please provide the correct args! " + this.getName());
            return;
        }

        if (args[0].equalsIgnoreCase(event.getPrefix())) {
            event.reply(event.getAuthor().getAsMention() + ", That prefix is already set.");
            return;
        }

        if (event.getPrefix().length() > 1) {
            event.reply(event.getAuthor().getAsMention() + ", Due to current odd issues, you cannot have a prefix longer than 1 character");
            return;
        }

        EmbedBuilder embedBuilder = new EmbedBuilder()
                .setTitle("Changed Lil' Ori Prefix")
                .setColor(Color.decode("#33539e"))
                .setFooter("Created by Oribuin", "https://imgur.com/ssJcsZg.png")
                .setDescription("**Old Prefix** " + event.getPrefix() +
                        "\n\n**New Prefix:** " + args[1]);

        event.reply(embedBuilder);
        this.bot.getGuildSettingsManager().updateGuild(event.getGuild(), args[1]);
        System.out.println(event.getAuthor().getAsTag() + " Updated \"" + event.getGuild().getName() + "\" Prefix to " + args[1]);

    }
}

File: src/main/kotlin/xyz/oribuin/lilori/commands/global/CmdPrefix.kt
package xyz.oribuin.lilori.commands.global

import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.Permission
import xyz.oribuin.lilori.LilOri
import xyz.oribuin.lilori.handler.Command
import xyz.oribuin.lilori.handler.CommandEvent
import java.awt.Color

class CmdPrefix(bot: LilOri) : Command(bot) {
    init {
        name = "Prefix"
        description = "Change the bot permission"
        aliases = emptyList()
        arguments = listOf("<prefix>")
        userPermissions = arrayOf(Permission.ADMINISTRATOR)
    }

    override fun executeCommand(event: CommandEvent) {
        val args = event.message.contentRaw.split(" ").toTypedArray()

        if (args.size == 1) {
            event.reply("${event.author.asMention}, Please provide the correct args!");
            return
        }

        if (args[0].equals(event.prefix, ignoreCase = true)) {
            event.reply(event.author.asMention + ", That prefix is already set.")
            return
        }


        if (event.prefix.length >= 2) {
            event.reply(event.author.asMention + ", Due to current odd issues, you cannot have a prefix longer than 1 character")
            return
        }

        val embedBuilder = EmbedBuilder()
                .setTitle("Changed Lil' Ori Prefix")
                .setColor(event.color)
                .setFooter("Created by Oribuin", "https://imgur.com/ssJcsZg.png")
                .setDescription("**Old Prefix** ${event.prefix}" +
                        " " +
                        "**New Prefix** ${args[1]}".trimIndent())

        event.reply(embedBuilder)
        bot.guildSettingsManager.updateGuild(event.guild, args[1])
        println(event.author.asTag + " Updated \"" + event.guild.name + "\" Prefix to " + args[1])
    }
}